splash-screen/android/src/main/java/com/capacitorjs/plugins/splashscreen/SplashScreen.java

```diff
@@ -59,6 +80,81 @@ public void showOnLaunch(final AppCompatActivity activity) {
         }
     }
 
+    /**
+     * Show the Splash Screen using the Android 12 API (31+)
+     * Uses Compat Library for backwards compatibility
+     *
+     * @param activity
+     * @param settings Settings used to show the Splash Screen
+     */
+    private void showWithAndroid12API(final AppCompatActivity activity, final SplashScreenSettings settings) {
+        if (activity == null || activity.isFinishing()) return;
+
+        activity.runOnUiThread(
+            () -> {
+                androidx.core.splashscreen.SplashScreen windowSplashScreen = androidx.core.splashscreen.SplashScreen.installSplashScreen(
+                    activity
+                );
+                windowSplashScreen.setKeepOnScreenCondition(() -> isVisible || isHiding);
+
+                // Set Fade Out Animation
+                windowSplashScreen.setOnExitAnimationListener(
+                    windowSplashScreenView -> {
+                        final ObjectAnimator fadeAnimator = ObjectAnimator.ofFloat(windowSplashScreenView.getView(), View.ALPHA, 1f, 0f);
+                        fadeAnimator.setInterpolator(new LinearInterpolator());
+                        fadeAnimator.setDuration(settings.getFadeOutDuration());
+
+                        fadeAnimator.addListener(
+                            new AnimatorListenerAdapter() {
+                                @Override
+                                public void onAnimationEnd(Animator animation) {
+                                    isHiding = false;
+                                    windowSplashScreenView.remove();
+                                }
+                            }
+                        );
+
+                        fadeAnimator.start();
+
+                        isHiding = true;
+                        isVisible = false;
+                    }
+                );
+
+                // Set Pre Draw Listener & Delay Drawing Until Duration Elapses
+                final View content = activity.findViewById(android.R.id.content);
```

Comment:
make this `content` variable global so it can be used in hide method instead of needing to pass an activity, since hide is public, adding a parameter would be a breaking change, so better avoid it.